File: email/validator.py
import re
import logging

logger = logging.getLogger(__name__)

EMAIL_PATTERN = re.compile(r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$')

#validate email address
def is_valid_email(email: str) -> bool:
    if not email or not isinstance(email, str):
        return False
    return bool(EMAIL_PATTERN.match(email.strip()))

#parse emails from text
def parse_emails(text: str) -> list:
    if not text or not isinstance(text, str):
        return []
    
    emails = []
    for line in text.strip().split('\n'):
        email = line.strip()
        if email and is_valid_email(email):
            emails.append(email)
    
    logger.info(f"Parsed {len(emails)} valid email(s) from input")
    return emails

File: email-utils/validator.js
import { logger } from '../config.js';

const EMAIL_PATTERN = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;

/**
 * Validate email address
 * @param {string} email - Email address to validate
 * @returns {boolean} - True if email is valid
 */
export function isValidEmail(email) {
    if (!email || typeof email !== 'string') {
        return false;
    }
    return EMAIL_PATTERN.test(email.trim());
}

/**
 * Parse emails from text (comma-separated or newline-separated)
 * @param {string} text - Text containing emails
 * @returns {string[]} - Array of valid email addresses
 */
export function parseEmails(text) {
    if (!text || typeof text !== 'string') {
        return [];
    }
    
    const emails = [];
    // Handle both comma-separated and newline-separated emails
    const lines = text.trim().split(/[,\n]/);
    
    for (const line of lines) {
        const email = line.trim();
        if (email && isValidEmail(email)) {
            emails.push(email);
        }
    }
    
    logger.info(`Parsed ${emails.length} valid email(s) from input`);
    return emails;
}

